felix/fv/flow_logs_goldmane_test.go

```diff
@@ -533,8 +523,82 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ goldmane flow log tests", [
 		}, "30s", "3s").ShouldNot(HaveOccurred())
 	}
 
-	It("should get expected flow logs", func() {
-		checkFlowLogs()
+	nflogRulesProgrammed := func() bool {
+		// nflogs are implemented as "-j NFLOG --nflog-group nlgroup" in iptables
+		out0, err := tc.Felixes[0].ExecOutput("iptables-save", "-t", "filter")
+		Expect(err).NotTo(HaveOccurred())
+		if strings.Count(out0, "NFLOG") != 0 || strings.Count(out0, "nflog-group") != 0 {
+			return true
+		}
+		out1, err := tc.Felixes[1].ExecOutput("iptables-save", "-t", "filter")
+		Expect(err).NotTo(HaveOccurred())
+		if strings.Count(out1, "NFLOG") != 0 || strings.Count(out1, "nflog-group") != 0 {
+			return true
+		}
+		return false
+	}
+	if NFTMode() {
+		nflogRulesProgrammed = func() bool {
+			// nflogs are implemented as "log prefix [prefix] group [nflog_group]" in nftables
+			out0, err := tc.Felixes[0].ExecOutput("nft", "list", "ruleset")
+			Expect(err).NotTo(HaveOccurred())
+			if strings.Count(out0, "log prefix") != 0 {
+				return true
+			}
+			out1, err := tc.Felixes[1].ExecOutput("nft", "list", "ruleset")
+			Expect(err).NotTo(HaveOccurred())
+			if strings.Count(out1, "log prefix") != 0 {
+				return true
+			}
+			return false
+		}
+	}
+
+	Context("with flow log disabled", func() {
+		BeforeEach(func() {
+			if BPFMode() {
+				Skip("nflog rules are irrelevent to bpf dataplane")
+			}
+
+			infra = getInfra()
+			opts = infrastructure.DefaultTopologyOptions()
+			opts.IPIPEnabled = false
+
+			testSetup()
+		})
+
+		It("should not program nflog rules", func() {
+			Eventually(nflogRulesProgrammed, "10s", "1s").Should(BeFalse(),
+				"Unexpected nflog rules to appear on the correct felix instances")
+			Consistently(nflogRulesProgrammed, "10s", "1s").Should(BeFalse(),
+				"Unexpected nflog rules to appear on the correct felix instances")
+		})
+	})
+
```

Comment:
Imo this test is redundant. It is already tested in UTs that it programs the rules or not. If anything (imo) it would make more sense to tests that it does not break connectivity and that policies work as expected in both cases if we have any such tests ....

Reply:
Yeah, and for that many UTs are added. This one is just to confirm the result in FV. The `testSetup()` function also checks the connectivity and also that the policies are programmed.

Reply:
Are you suggesting to remove it?

Reply:
With current situation, it seems there is no need for this test as stated. In all other FVs, flow logs are disabled, and as such connectivity and policies are tested.